GeneratorInterface/Pythia8Interface/src/Py8HMC3InterfaceBase.cc

```diff
@@ -0,0 +1,232 @@
+#include <memory>
+#include <string>
+
+#include "GeneratorInterface/Pythia8Interface/interface/Py8HMC3InterfaceBase.h"
+#include "GeneratorInterface/Pythia8Interface/interface/SLHAReaderBase.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <filesystem>
+
+// EvtGen plugin
+//
+//#include "Pythia8Plugins/EvtGen.h"
+
+using namespace Pythia8;
+
+namespace gen {
+
+  Py8HMC3InterfaceBase::Py8HMC3InterfaceBase(edm::ParameterSet const& ps)
+      : BaseHadronizer(ps), useEvtGen(false), evtgenDecays(nullptr) {
+    fParameters = ps;
+
+    pythiaPylistVerbosity = ps.getUntrackedParameter<int>("pythiaPylistVerbosity", 0);
```

Comment:
If these parameters get initialized in the class initializing list, they can be better declared as `const` in the header